src/test/java/org/opensearch/ad/ml/ModelManagerTests.java

```diff
@@ -973,4 +987,19 @@ public void getPreviewResults_returnAnomalies_forLastAnomaly() {
     public void getPreviewResults_throwIllegalArgument_forInvalidInput() {
         modelManager.getPreviewResults(new double[0][0], shingleSize);
     }
+
+    @Test
+    public void getAnomalyResultForEntity_withercf() {
+        AnomalyDescriptor anomalyDescriptor = new AnomalyDescriptor();
+        anomalyDescriptor.setRcfScore(2);
+        anomalyDescriptor.setAnomalyGrade(1);
+        when(this.ercf.process(this.point)).thenReturn(anomalyDescriptor);
+
+        ThresholdingResult result = modelManager
+            .getAnomalyResultForEntity(this.point, this.modelState, this.detectorId, this.anomalyDetector, null);
+        assertEquals(
+            new ThresholdingResult(anomalyDescriptor.getAnomalyGrade(), /*TODO: pending ercf*/1.0, anomalyDescriptor.getRcfScore()),
+            result
+        );
+    }
```

Comment:
Instead of hardcoding the value 2 for the RCF score and 1 for the anomaly grade, this test function could accept arguments for the RCF score and anomaly grade. Then you can test a few different combinations of values, including edge cases where an unexpected behavior could happen.

Reply:
What is `/*TODO: pending ercf*/`? Is that waiting on another PR?

Reply:
1. Parameterized testing is useful. In this case, the logic is quite simple as just copying data so the test can therefore be simple. If the logic becomes more complex, the corresponding tests can be parameterized to scale to the test cases. However, that is not always needed.
2. The comment is a TODO item for the confidence part of ercf is still in development and not available yet. It unblocks development and integration on the plugin side for the time being.